### PR TITLE
feat: add option to broadcast rpc psbt

### LINKED ISSUE #3895: Add option to broadcast PSBTs upon finalization
Instead of always returning to the app for broadcasting. I've heard a couple developers request this feature recently.

### PR DESCRIPTION
**Pull request by fbwoolf**
> Try out this version of the Hiro Wallet - download [extension builds](https://github.com/hirosystems/wallet/actions/runs/5766015966).

This PR adds a param to the `signPsbt` rpc request so an app can optionally have us finalize and broadcast the transaction.

*Not sure if the param name is great? `broadcast`? Doesn't seem to suggest a boolean?

Also, I took the time here to add a `PsbtSignerContext` to remove some of the prop passing and refactored the folder nesting a bit.

### CODE REVIEW

File: src/app/features/psbt-signer/psbt-signer.context.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  addressNativeSegwit: '',
+  addressTaproot: '',
+  addressNativeSegwitTotal: createMoney(0, 'BTC'),
+  addressTaprootTotal: createMoney(0, 'BTC'),
+  fee: createMoney(0, 'BTC'),
+  isPsbtMutable: false,
+  psbtInputs: [],
+  psbtOutputs: [],
+  shouldDefaultToAdvancedView: false,
+});
```
Comment (kyranjamie) [inline src/app/features/psbt-signer/psbt-signer.context.ts]: Is this legitimately not defined in many cases at start of flow, or is the mock data just to satisfy type?

If latter, suggest a default of `null`, a custom hook, and a `throw` if `null` so you can use `usePsbtSignerContext()` knowing data is defined

File: src/app/features/psbt-signer/psbt-signer.tsx [hunk 2]
```diff
@@ -39,15 +49,27 @@ function getPsbtTxOutputs(psbtTx: btc.Transaction) {
...
+    indexesToSign,
+    isBroadcasting,
+    name,
+    origin,
+    onCancel,
+    onSignPsbt,
+    psbtHex,
+  } = props;
   const navigate = useNavigate();
+  const addressNativeSegwit = useCurrentAccountNativeSegwitIndexZeroSigner().address;
```
Comment (kyranjamie) [inline src/app/features/psbt-signer/psbt-signer.tsx]: destructure here too?

**Review by alter-eggo — APPROVED** [review]
`isPsbtBroadcast`?

**Comment by fbwoolf** [comment]
@kyranjamie if you are arnd before the wkend, can you review the changes asked for here?

**Review by kyranjamie — APPROVED** [review]
LGTM

**Pull request merged**